clang/lib/CodeGen/CGOpenMPRuntimeGPU.cpp

```diff
@@ -3362,6 +3362,8 @@ Address CGOpenMPRuntimeGPU::getAddressOfLocalVariable(CodeGenFunction &CGF,
       break;
     case OMPAllocateDeclAttr::OMPLargeCapMemAlloc:
     case OMPAllocateDeclAttr::OMPCGroupMemAlloc:
+      if (VD->hasGlobalStorage()) 
+        AS = getLangASFromTargetAS(CGF.CGM.getModule().getDataLayout().getDefaultGlobalsAddressSpace());
```

Comment:
This function is only for globals, so we don't need to check VD after all.

Reply:
I removed the global variable check